tests/parallel_scaling/scaling.py

```diff
@@ -29,6 +31,47 @@
 }
 
 
+class HPCDetails:
+    batch_templates = {
+        "gadi": "qsub -v GADOPT_CHECKOUT={gadopt_checkout},GADOPT_SETUP={gadopt_setup} -W block=true -N scaling_{level} -l storage=gdata/xd2+scratch/xd2+gdata/fp50,ncpus={cores},walltime=04:00:00,mem={mem}GB,wd,jobfs=200GB -q normalsr -P {project} -o pbs_output/l{level}.out -e pbs_output/l{level}.err -- ./run_gadi.sh {level}",
+        "setonix": "sbatch --export GADOPT_CHECKOUT={gadopt_checkout},GADOPT_SETUP={gadopt_setup} --wait -J scaling_{level} --exclusive --ntasks={cores} --nodes={nodes} -t 4:00:00 -p work -A {project} -o slurm_output/l{level}.out -o slurm_output/l{level}.err -- ./run_setonix.sh {level}",
+    }
+    required_environment = {
+        "gadi": {"gadopt_checkout", "gadopt_setup", "project"},
+        "setonix": {"gadopt_checkout", "gadopt_setup", "project"},
+    }
+
+    @staticmethod
+    def hpc_system_is_gadi():
+        return socket.gethostname().startswith("gadi")
+
+    @staticmethod
+    def hpc_system_is_setonix():
+        if os.path.isfile("/software/pawsey/motd"):
+            with open("/software/pawsey/motd") as f:
+                for line in f.readlines():
+                    if line == "     ___  ___| |_ ___  _ __ (_)_  __           ,########(,,  ...     ,########/,\n":
+                        return True
+            return False
+        else:
+            return False
```

Comment:
Would something like `socket.getfqdn()` or `socket.gethostbyaddr(socket.gethostname())` help streamline this block? What about `echo $HOSTNAME` or `hostnamectl` if not? All these commands seem to return information that includes the string `setonix`.